code/authenticators/LDAPLoginForm.php

```diff
@@ -29,22 +34,40 @@ class LDAPLoginForm extends MemberLoginForm {
 	public function __construct($controller, $name, $fields = null, $actions = null, $checkCurrentUser = true) {
 		parent::__construct($controller, $name, $fields, $actions, $checkCurrentUser);
 
-		$emailField = new TextField('Username', _t('Member.USERNAME', 'Username'));
+		// will be used to get correct Link()
+		$this->ldapSecController = Injector::inst()->create('LDAPSecurityController');
```

Comment:
True, there are quite a lot of copy paste and stuff like from the previous version. It's hard finding that balance of only overriding what needed (easier upgrades) and customizing it.

Dunno in this case.